pyatlan/model/typedef.py

```diff
@@ -390,6 +432,158 @@ def create(
     )
     is_default_value_null: Optional[bool] = Field(description="TBC")
 
+    def __setattr__(self, name, value):
+        if name in AttributeDef._convenience_properties:
+            return object.__setattr__(self, name, value)
+        super().__setattr__(name, value)
+
+    _convenience_properties: ClassVar[list[str]] = [
+        "applicable_asset_types",
+        "applicable_glossary_types",
+        "applicable_other_asset_types",
+        "applicable_entity_types",
+        "applicable_connections",
+        "applicable_glossaries",
+    ]
+
+    @property
+    def applicable_entity_types(self) -> set[str]:
+        """
+        Set of entities on which this attribute can be applied.
+        Note: generally this should be left as-is. Any overrides should instead be applied through
+        one or more of applicable_asset_types, applicable_glossary_types, or applicable_other_asset_types.
+        """
+        if self.options and self.options.applicable_entity_types:
+            return set(json.loads(self.options.applicable_entity_types))
+        return set()
+
+    @applicable_entity_types.setter
+    def applicable_entity_types(self, entity_types: set[str]):
```

Comment:
Nice — so users can still send in a set, even though it's marshalled into JSON as a string 👍

Reply:
Done